File: FirstClass/problem20.m
t = linspace (-pi, pi, 1000);
r = linspace (-2*pi, 2*pi, 1000);

x = 4*cos(r).*sec (t);
y = 2*sin(r).*sin (t);
z = tan(t);

plot3 (t, r, x, y, z, "linewidth", 2);

File: FirstClass/Python/problem_20.py
#La diferencia de las graficas se debe a la cantidad de puntos
#Si disminuye el 100 a 50, la grafica se aproximara mas a lo hecho en octave
import numpy as np 
import matplotlib.pyplot as plt 
import math
if __name__ == '__main__':
    fig = plt.figure ()
    ax = fig.gca (projection = '3d')
    tx = np.linspace (-8,8, 100)
    ty = np.linspace (-8,8, 100)
    [xx, yy] = np.meshgrid (tx, ty)
    X = 4 * np.cos (yy) * np.sin (xx)
    Y = 2 * np.sin (yy) * np.sin (xx)
    Z = np.tan (xx)
    ax.plot_surface (X, Y, Z)
    plt.show()